src/cmd/tools/lint.go

```diff
@@ -26,32 +26,52 @@ func init() {
 		PersistentPreRun: func(cmd *cobra.Command, args []string) {
 			config.SkipLogFile = true
 		},
-		Long:    "Validate an OSCAL document is properly configured against the OSCAL schema",
+		Long:    "Validate OSCAL documents are properly configured against the OSCAL schema",
 		Example: lintHelp,
 		Run: func(cmd *cobra.Command, args []string) {
-			spinner := message.NewProgressSpinner("Linting %s", opts.InputFile)
-			defer spinner.Stop()
+			var validationResults []validation.ValidationResult
+			var errorsOccurred bool
 
-			validationResp, err := validation.ValidationCommand(opts.InputFile)
+			for _, inputFile := range opts.InputFiles {
+				spinner := message.NewProgressSpinner("Linting %s\n", inputFile)
+				defer spinner.Stop()
 
-			for _, warning := range validationResp.Warnings {
-				message.Warn(warning)
+				validationResp, err := validation.ValidationCommand(inputFile)
+
+
+				if err != nil {
+					message.Warnf("Failed to lint %s: %v\n", inputFile, err)
+					errorsOccurred = true
+					continue
+				}
+
+				for _, warning := range validationResp.Warnings {
+					message.Warn(warning)
+				}
+
+				validationResults = append(validationResults, validationResp.Result)
+
+				message.Infof("Successfully validated %s is valid OSCAL version %s %s\n", inputFile, validationResp.Validator.GetSchemaVersion(), validationResp.Validator.GetModelType())
+				spinner.Success()
 			}
 
 			if opts.ResultFile != "" {
-				validation.WriteValidationResult(validationResp.Result, opts.ResultFile)
+				err := validation.WriteValidationResults(validationResults, opts.ResultFile)
+				if err != nil {
+					message.Fatalf(err, "Failed to write validation results to %s\n", opts.ResultFile)
```

Comment:
```suggestion
					message.Fatalf(err, "Failed to write validation results to %s with error: %s", opts.ResultFile, err.Error())
```

This adds more context around why the error happened -IE if I supplied `-r output.xml` it will tell me that it failed to write AND that the file type can only be yaml or json.